instrumentation/net/http/otelhttp/transport.go

```diff
@@ -109,7 +148,22 @@ func (t *Transport) RoundTrip(r *http.Request) (*http.Response, error) {
 		ctx = httptrace.WithClientTrace(ctx, t.clientTrace(ctx))
 	}
 
+	labeler := &Labeler{}
+	ctx = injectLabeler(ctx, labeler)
+
 	r = r.Clone(ctx) // According to RoundTripper spec, we shouldn't modify the origin request.
+
+	writeRecordFunc := func(int64) {}
+	var bw bodyWrapper
+	// if request body is nil or NoBody, we don't want to mutate the body as it
+	// will affect the identity of it in an unforeseeable way because we assert
+	// ReadCloser fulfills a certain interface and it is indeed nil or NoBody.
+	if r.Body != nil && r.Body != http.NoBody {
+		bw.ReadCloser = r.Body
+		bw.record = writeRecordFunc
+		r.Body = &bw
+	}
```

Comment:
Why are we wrapping the body, with a recorder that does nothing?

Reply:
I'm saying that if we don't do anything in the record method, we shouldn't wrap it.  
That is to allow us to record a metric of how much was read.

Reply:
apologies, you're absolutely right. this is some left over cruft from the old PR where it was being used to record events. Will remove this shortly.

Reply:
We aren't doing anything in the record _callback_, however we are still using the wrapper to pull the number of bytes that were read.